### PR TITLE
OD model and data util functions in rai_test_utils

### PR DESCRIPTION
**Pull request by Advitya17**
## Checklist

- [ ] I have added screenshots above for all UI changes.
- [ ] I have added e2e tests for all UI changes.
- [ ] Documentation was updated if it was needed.

### CODE REVIEW

File: rai_test_utils/requirements.txt [hunk 1]
```diff
@@ -5,3 +5,5 @@ scipy
 shap
 lightgbm
 xgboost
+torch==1.13.1
+torchvision==0.14.1
```
Comment (gaugup) [inline rai_test_utils/requirements.txt:9]: Could you maybe add these dependencies as extras for his package? May be add extra like object_detection. `pip install rai_test_utils[object_detection]`.

More on extras here:- https://stackoverflow.com/questions/52474931/what-is-extra-in-pypi-dependency

File: rai_test_utils/rai_test_utils/models/torch/torch_model_utils.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import os
+import urllib.request as request_file
+
+import torch
+import torchvision
+from torchvision.models.detection.faster_rcnn import FastRCNNPredictor
+
+
+# download fine-tuned recycling model from url
+def download_assets(filepath, force=False):
```
Comment (gaugup) [inline rai_test_utils/rai_test_utils/models/torch/torch_model_utils.py:13]: Could you add doc strings for all the functions in this file?

File: rai_test_utils/rai_test_utils/datasets/vision/object_detection_data_utils.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+import os
+import urllib.request as request_file
+import xml.etree.ElementTree as ET
+from zipfile import ZipFile
+
+import pandas as pd
+
+
+def load_fridge_object_detection_dataset_labels():
+
```
Comment (gaugup) [inline rai_test_utils/rai_test_utils/datasets/vision/object_detection_data_utils.py:13]: Could you add doc strings for all the functions in this file?

File: .github/workflows/CI-python.yml [hunk 4]
```diff
@@ -20,7 +20,7 @@ jobs:
           ["responsibleai", "erroranalysis", "raiutils", "rai_test_utils"]
         operatingSystem:
           [ubuntu-20.04, ubuntu-latest, macos-latest, windows-latest]
-        pythonVersion: ["3.6", "3.7", "3.8", "3.9", "3.10"]
```
Comment (gaugup) [inline .github/workflows/CI-python.yml:23]: The change to remove 3.6 could be committed as a separate PR. Might be good for tracking purpose.

Comment (Advitya17) [reply .github/workflows/CI-python.yml:23]: Sure - https://github.com/microsoft/responsible-ai-toolbox/pull/2249

File: rai_test_utils/rai_test_utils/datasets/vision/__init__.py [hunk 5]
```diff
@@ -0,0 +1,10 @@
+# Copyright (c) Microsoft Corporation
+# Licensed under the MIT License.
+
+"""Namespace for tabular datasets."""
```
Comment (romanlutz) [inline rai_test_utils/rai_test_utils/datasets/vision/__init__.py:4]:
```suggestion
"""Namespace for vision datasets."""
```

File: rai_test_utils/rai_test_utils/datasets/vision/object_detection_data_utils.py [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    for _, filename in enumerate(os.listdir(annotations_folder)):
+        if filename.endswith(".xml"):
+            print("Parsing " + os.path.join(src_images, filename))
+
+            root = ET.parse(
+                os.path.join(annotations_folder, filename)
+            ).getroot()
+
+            # use if needed
+            # width = int(root.find("size/width").text)
```
Comment (romanlutz) [inline rai_test_utils/rai_test_utils/datasets/vision/object_detection_data_utils.py:37]: How do I know if it's needed?

**Review by romanlutz — APPROVED** [review]

File: rai_test_utils/rai_test_utils/utilities/utils.py [hunk 7]
```diff
@@ -62,3 +73,113 @@ def retrieve_dataset(dataset, **kwargs):
...
+        except ValueError:
+            # if images of different sizes, try to convert one by one
+            jagged = np.empty(len(dataset), dtype=object)
+            for i, x in enumerate(dataset):
+                jagged[i] = get_image_from_path(x, image_mode)
+            dataset = jagged
+    return dataset
+
+
+def get_images(dataset, image_mode, transformations=None):
```
Comment (gaugup) [inline rai_test_utils/rai_test_utils/utilities/utils.py:146]: Could this method and its dependencies live in dataset/vision? Looks like these methods are specific to vision so not sure if these should be here.

File: rai_test_utils/setup.py [hunk 8]
```diff
@@ -15,6 +15,13 @@
 with open('requirements.txt') as f:
     install_requires = [line.strip() for line in f]
 
+extras_require = {
+    "object_detection": [
+        "torch==1.13.1",
+        "torchvision==0.14.1",
+    ]
+}
+
```
Comment (gaugup) [inline rai_test_utils/setup.py:24]: Thanks for creating the extras.

File: rai_test_utils/rai_test_utils/models/torch/torch_model_utils.py [hunk 9]
```diff
@@ -0,0 +1,72 @@
+# Copyright (c) Microsoft Corporation
+# Licensed under the MIT License.
+
+import os
+import urllib.request as request_file
+
+import torch
```
Comment (gaugup) [inline rai_test_utils/rai_test_utils/models/torch/torch_model_utils.py:7]: I think this code (7-9) should be under try/catch as the import would fail if torch is not installed (not it might be installed as it is an extra.).

**Review by gaugup — CHANGES_REQUESTED** [review]

File: rai_test_utils/setup.py (hunk 8 above)
Comment (gaugup) [reply rai_test_utils/setup.py:24]: do you mind moving them in requirements-object-detection.txt in this package and then reading from it.

File: rai_test_utils/rai_test_utils/datasets/vision/object_detection_data_utils.py (hunk 6 above)
Comment (Advitya17) [reply rai_test_utils/rai_test_utils/datasets/vision/object_detection_data_utils.py:37]: Lines 50-53 mention it, in case the coordinates need to be normalized.

File: rai_test_utils/requirements.txt [hunk 10]
```diff
@@ -4,4 +4,4 @@ scikit-learn
 scipy
 shap
 lightgbm
-xgboost
+xgboost
```
Comment (gaugup) [inline rai_test_utils/requirements.txt:7]: What is different in this line?

**Review by gaugup — APPROVED** [review]

File: rai_test_utils/requirements.txt (hunk 10 above)
Comment (Advitya17) [reply rai_test_utils/requirements.txt]: Honestly I had the same question lol, looks like a newline tweak, so restored it from main.
